src/main/java/sparksoniq/jsoniq/item/EmptySequenceItem.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Stefan Irimescu
+ *
+ */
+ package sparksoniq.jsoniq.item;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import sparksoniq.jsoniq.item.metadata.ItemMetadata;
+
+import javax.naming.OperationNotSupportedException;
+import java.math.BigDecimal;
+
+public class EmptySequenceItem extends AtomicItem {
```

Comment:
We would need to discuss this. Such an item does not exist. The correct way to implement an empty sequence returned by an iterator is to make sure that hasNext() returns false -- there is no need for extra complexity.

Reply:
Yes. I understand that this is tricky, but it is simpler than it seems. A null atomic value is really something fundamentally different: a sequence of one null item. An empty sequence is not an item: it is the absence of any item in the first place.

I will clarify tomorrow if needed.

Reply:
The EmptySequenceIterator should not return EmptySequenceItem. It should simply have hasNext() return false, and next() return an error. I recommend resisting the temptation to "copy" the behavior of null.